File: CandyRainTolstonogova/CandyRainC++/CandyRain/Player.cpp
#include "Player.h"
#include <iostream> 
#include "GameObject.h"

Player::Player(int x, int y, const std::string& sprite_path, int spd, bool alive) :
    GameObject(x, y, sprite_path), speed(spd), life(alive) {
    std::cout << "Объект Player создан" << std::endl;
}

Player::~Player() {
    std::cout << "Объект Player уничтожен" << std::endl;
}
void Player::moveRight() {
    int new_position = getpositionX();
    new_position += speed;
    setpositionX(new_position);
    std::cout << "Player перемещен вправо. Новая позиция X: " << getpositionX() << std::endl; 
}

void Player::moveLeft() {
    int new_position = getpositionX();
    new_position -= speed;
    setpositionX(new_position);
    std::cout << "Player перемещен влево. Новая позиция X: " << getpositionX() << std::endl;
}

// Getter
int Player::getlife() const { return life; }

// Setter
void Player::setlife(int count) {life = count; }

File: CandyRainTolstonogova/CandyRainJava/Player.java
public class Player extends GameObject {
    private int speed;
    private boolean life;
    
    public Player(int x, int y, String spritePath, int spd, boolean alive) {
        super(x, y, spritePath);
        this.speed = spd;
        this.life = alive;
        System.out.println("Объект Player создан");
    }
    
    public void moveRight() {
        int newPosition = getPositionX();
        newPosition += speed;
        setPositionX(newPosition);
        System.out.println("Player перемещен вправо. Новая позиция X: " + getPositionX());
    }
    
    public void moveLeft() {
        int newPosition = getPositionX();
        newPosition -= speed;
        setPositionX(newPosition);
        System.out.println("Player перемещен влево. Новая позиция X: " + getPositionX());
    }
    
    // Getter and Setter
    public boolean getLife() { return life; }
    public void setLife(boolean life) { this.life = life; }
}